projects/smart-table/src/lib/smart-table/smart-table.component.ts

```diff
@@ -23,41 +21,51 @@ import {
   SmartTableOptions,
   UpdateFilterArgs,
 } from './smart-table.types';
+import {first, map, tap} from 'rxjs/operators';
+import {PROVIDE_ID} from '../indentifier.provider';
+import {Observable, of} from 'rxjs';
 
 @Component({
   selector: 'aui-smart-table',
   styleUrls: ['./smart-table.component.scss'],
   templateUrl: './smart-table.component.html'
 })
-export class SmartTableComponent implements AfterViewInit {
+export class SmartTableComponent {
   @Input() apiUrl: string;
   @Input() httpHeaders: HttpHeaders;
   @Input() columnTypes: SmartTableColumnCustomType[] = [];
+
   @Input()
   set configuration(configuration: SmartTableConfig) {
-    this._configuration = configuration;
-    if (Array.isArray(configuration.columns) && configuration.columns.length) {
-      this.baseFilters = configuration.baseFilters || [];
-
-      if (configuration.options) {
-        this.options = {
-          ...SMARTTABLE_DEFAULT_OPTIONS,
-          ...configuration.options
+    const id: string = configuration && configuration.options && configuration.options.storageIdentifier;
+    this.getConfiguration(id).pipe(
+      first(),
+      map(config => {
+        return {
+          ...config,
+          ...configuration,
```

Comment:
De code lijkt juist, maar kan qua naamgeving mss nog wel verbeterd worden… je ziet maar (aangezien je zelf al aangaf dat het niet de properste code is).